flanker/addresslib/address.py

```diff
@@ -558,7 +558,7 @@ def to_unicode(self):
         Converts to unicode.
         """
         if self.display_name:
-            return u'{} <{}@{}>'.format(self.display_name, self.local_part, self.domain)
+            return u'{} <{}@{}>'.format(smart_quote(self.display_name), self.local_part, self.domain)
```

Comment:
`encode_string` actually does that. `smart_quote` only adds the double quotes and escapes any `"` or `\` characters.